tt02-test/Makefile

```diff
@@ -1,2 +1,6 @@
 PATTERN = demo
 include ../Makefile
+
+submodule-all: $(PATTERN).elf $(PATTERN).bin $(PATTERN).hex flash_tigard
```

Comment:
could you comment on this?

Reply:
Because the target `all` exists in the included `../Makefile` it was easier to indirect the goal into a sub-target `submodule-all`.

Because the SUBMODULES variable is only setup if you are the superrior make process executing from the toplevel project directory.  The included `for x in $SUBMODULES` loop effectively runs but does nothing in the context of the subdir.

Now the subdir Makefile then provides its own default target goals with the line 4 quoted above.